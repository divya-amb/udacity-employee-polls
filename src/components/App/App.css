.App-container {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
}

.App-content {
  flex: 1;
  margin: 5rem;
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
}
