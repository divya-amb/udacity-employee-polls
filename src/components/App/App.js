import { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { Routes, Route } from "react-router-dom";
import LoadingBar from "react-redux-loading-bar";
import { handleInitialData } from "../../actions/shared";
import Login from "../Login";
import NavBar from "../NavBar";
import Home from "../Home";
import Leaderboard from "../Leaderboard";
import AddQuestion from "../AddQuestion";
import QuestionDetails from "../QuestionDetails";
import NotFound from "../NotFound";
import Loading from "../Loading";
import "./App.css";

const App = ({authedUserID, loading, dispatch}) => {
  useEffect(() => {
    if (authedUserID) {
      dispatch(handleInitialData());
    }
  }, [authedUserID]);

  return (
    <Fragment>
      <LoadingBar />
        <div className="App-container">
          <NavBar />
          {authedUserID ? (
              <div className="App-content">
                {loading ? <Loading /> : (<Routes>
                  <Route path="/questions/:qid" element={<QuestionDetails />} />
                  <Route path="/leaderboard" element={<Leaderboard />} />
                  <Route path="/add" element={<AddQuestion />} />
                  <Route path="/" exact element={<Home />} />
                  <Route path="*" element={<NotFound />} />
                </Routes>)}
              </div>
          ) : (
            <Routes>
              <Route path="*" element={<Login />} />
            </Routes>
          )}
        </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser, users = {}, questions = {} }) => ({
  authedUserID: authedUser.id,
  loading: Object.keys(users).length === 0 || Object.keys(questions).length === 0,
});

export default connect(mapStateToProps)(App);