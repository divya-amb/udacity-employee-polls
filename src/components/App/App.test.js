import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from "redux";
import reducer from "../../reducers";
import middleware from "../../middleware";
import { BrowserRouter as Router } from "react-router-dom";
import App from './App';

describe('App', () => {
  it('will match snapshot', () => {
    const store = createStore(reducer, middleware);
    const view = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
  
    expect(view).toMatchSnapshot();
  });

  it('will render the Employee Polls application', () => {
    const store = createStore(reducer, middleware);
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
  
    expect(screen.getByText("Employee Polls")).toBeInTheDocument();
  });
});
