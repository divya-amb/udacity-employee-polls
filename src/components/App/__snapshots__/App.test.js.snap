// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="App-container"
      >
        <form>
          <div
            class="NavBar-container"
          >
            <div
              class="NavBar-header"
            >
              <h1>
                Employee Polls
              </h1>
            </div>
          </div>
        </form>
        <form>
          <div
            class="Login-container"
          >
            <h3>
              Login
            </h3>
            <div
              class="Login-inputDiv"
            >
              <span>
                Username: 
              </span>
              <input
                data-testid="username-input"
                type="text"
                value=""
              />
            </div>
            <div
              class="Login-inputDiv"
            >
              <span>
                Password: 
              </span>
              <input
                data-testid="password-input"
                type="password"
                value=""
              />
            </div>
            <button
              data-testid="submit-button"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="App-container"
    >
      <form>
        <div
          class="NavBar-container"
        >
          <div
            class="NavBar-header"
          >
            <h1>
              Employee Polls
            </h1>
          </div>
        </div>
      </form>
      <form>
        <div
          class="Login-container"
        >
          <h3>
            Login
          </h3>
          <div
            class="Login-inputDiv"
          >
            <span>
              Username: 
            </span>
            <input
              data-testid="username-input"
              type="text"
              value=""
            />
          </div>
          <div
            class="Login-inputDiv"
          >
            <span>
              Password: 
            </span>
            <input
              data-testid="password-input"
              type="password"
              value=""
            />
          </div>
          <button
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
